runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -357,9 +366,14 @@ function LanguageTree:_parse_regions(range)
       )
     then
       self._parser:set_included_ranges(ranges)
+      self._parser:set_timeout(timeout and timeout * 1000 or 0) -- ms -> micros
       local parse_time, tree, tree_changes =
         tcall(self._parser.parse, self._parser, self._trees[i], self._source, true)
```

Comment:
Does this work for multiple trees?

I looked into tree-sitter and it doesn't change the tree when previous parse didn't finish. So I assume this would parse 1 tree n times instead of parsing n trees once.